### PR TITLE
[REVIEW] Removed `move` from self-defined and replaced with standard library `move`

### LINKED ISSUE #586: [FEA] Replace `move` with standard library `move`
**Is your feature request related to a problem? Please describe.**
Cython backported `std::move` to 0.29, and so RMM could use the `move` from the standard library instead of from self-defined.

**Describe the solution you'd like**
Similar to [this in cuDF](https://github.com/rapidsai/cudf/pull/6383)

### PR DESCRIPTION
**Pull request by isVoid**
Closes #586

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@
 - PR #579 CMake: handle thrust via target
 - PR #581 Improve logging documentation
 - PR #585 Update ci/local/README.md
+- PR #586 Replaced `move` with `std::move`
```
Comment (jakirkham) [inline CHANGELOG.md:31]: Once we fix this, CI should be able to start

```suggestion
- PR #587 Replaced `move` with `std::move`
```

Comment (isVoid) [reply CHANGELOG.md]: Ah yes, bad typos!

**Review by kkraus14 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jakirkham) [reply CHANGELOG.md]: No worries. This happens to me too as I try to guess what the number will be before submitting 😄

**Pull request merged**